app/javascript/components/supplemental_application/actions.js

```diff
@@ -1,11 +1,31 @@
 import apiService from '~/apiService'
 import domainToApi from '~/components/mappers/domainToApi'
 import Alerts from '~/components/Alerts'
+import { isEmpty } from 'lodash'
+
+export const updateApplication = async (application) => {
+  const leasePromise = updateLease(
+    application['lease'], application['primaryApplicantContact'], application['id']
+  )
 
-export const updateApplicationAction = async (application) => {
   const applicationApi = domainToApi.buildApplicationShape(application)
-  const response = await apiService.submitApplication(applicationApi)
-  return response
+  const appPromise = apiService.submitApplication(applicationApi)
+
+  const [appResponse, leaseResponse] = await Promise.all([appPromise, leasePromise])
+  console.log('returning with a response', appResponse, leaseResponse)
+  return appResponse !== false && leaseResponse !== false
+}
+
+const updateLease = async (lease, primaryApplicantContact, applicationId) => {
+  if (!isEmpty(lease)) {
+    let leaseApi = domainToApi.mapLease(lease)
+    // TODO: We should consider setting the Tenant on a Lease more explicitly
+    // either via a non-interactable form element or using Salesforce
+    leaseApi['primary_applicant_contact'] = primaryApplicantContact
```

Comment:
this is super duper awkward. I'm working on figuring out what the long term plan is for this Tenant field so we can come up with a better solution.